component/discovery/consulagent/consulagent.go

```diff
@@ -0,0 +1,85 @@
+package consulagent
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/river/rivertypes"
+	promcfg "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.consulagent",
+		Args:    Arguments{},
+		Exports: discovery.Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+type Arguments struct {
+	Server          string            `river:"server,attr,optional"`
+	Token           rivertypes.Secret `river:"token,attr,optional"`
+	Datacenter      string            `river:"datacenter,attr,optional"`
+	TagSeparator    string            `river:"tag_separator,attr,optional"`
+	Scheme          string            `river:"scheme,attr,optional"`
+	Username        string            `river:"username,attr,optional"`
+	Password        rivertypes.Secret `river:"password,attr,optional"`
+	RefreshInterval time.Duration     `river:"refresh_interval,attr,optional"`
```

Comment:
I don't see a use for this field in the upstream configuration, do we need this?

Reply:
Do you mean RefreshInterval? In the promtail implementation the RefreshInterval is used in the Run function to create a ticker which checks repeatedly for new services to watch